packages/next/src/server/base-server.ts

```diff
@@ -2422,15 +2421,24 @@ export default abstract class Server<ServerOptions extends Options = Options> {
       }
     }
     const { res, query } = ctx
+    const isNextPrivateUnmatchedRoute = !!query.__nextPrivateUnmatchedRoute
+    delete query.__nextPrivateUnmatchedRoute
+
     try {
       let result: null | FindComponentsResult = null
 
       const is404 = res.statusCode === 404
       let using404Page = false
 
+      const isStandaloneNextPrivate404Routing =
+        ctx.renderOpts.nextConfigOutput === 'standalone' &&
+        isNextPrivateUnmatchedRoute
+
       // use static 404 page if available and is 404 response
       if (is404) {
-        if (this.hasAppDir) {
+        // If it's standalone mode rendering unmatched `_next` routes, it's still in initial routing worker,
+        // Use pages 404 instead of app dir not-found page to avoid access RSC rendering where require-hook is not set up.
```

Comment:
Does it mean that non-matching `/_next/*` routes will be rendered with 404.js from pages?

Reply:
They're failed from that call, for standalone server the static files match is quite different that the server doesn't serve them? correct me if I get it wrong @ijjk